staging/src/k8s.io/apiserver/pkg/storage/cacher/caching_object.go

```diff
@@ -214,6 +231,10 @@ func (o *cachingObject) conditionalSet(isNoop func() bool, set func()) {
 	if isNoop() {
 		return
 	}
+	if !o.deepCopied {
+		o.object = o.object.DeepCopyObject().(metaRuntimeInterface)
+		o.deepCopied = false
```

Comment:
shouldn't this be `o.deepCopied = true`?

Reply:
Of couse it should be - thanks for catching this.

Fixed and added a test.